### PR TITLE
Add TCP option for dig plugin.

### PR DESCRIPTION
**Pull request by k-e0**
##### SUMMARY

Add TCP option for dig plugin.

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

dig plugin

### CODE REVIEW

**Comment by jpmens** [comment]
That's excellent, @k-e0, thank you -- very useful!

File: changelogs/fragments/7343-dig-tcp-option.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - dig - add TCP option to enable the use of TCP connection during dns lookup. (https://github.com/ansible-collections/community.general/pull/7343).
```
Comment (felixfontein) [inline changelogs/fragments/7343-dig-tcp-option.yml:2]:
```suggestion
  - dig lookup plugin - add TCP option to enable the use of TCP connection during DNS lookup (https://github.com/ansible-collections/community.general/pull/7343).
```

File: plugins/lookup/dig.py [hunk 2]
```diff
@@ -70,6 +70,10 @@
           - "Class."
         type: str
         default: 'IN'
+      tcp:
+        description: Use TCP to lookup DNS records.
+        default: false
+        type: bool
```
Comment (felixfontein) [inline plugins/lookup/dig.py:76]:
```suggestion
        type: bool
        version_added: 7.5.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! I've got some small formal details that need to be fixed, then we can merge :)

**Comment by k-e0** [comment]
> Thanks for your contribution! I've got some small formal details that need to be fixed, then we can merge :)

Thank you for the review! I've committed your suggestions.

**Comment by jpmens** [comment]
shipit

**Comment by felixfontein** [comment]
@k-e0 thanks for your contribution!
@jpmens thanks for reviewing!

**Pull request merged**